modules/targetAcquisition/taxi/TaxiWorker.py

```diff
@@ -9,9 +9,8 @@ def taxi_worker(pause, exitRequest, pipelineIn, pipelineOut):
         pause.release()
         
         frame = pipelineIn.get()
-        taxi.main(frame)
-
-        pipelineOut.put()
+        if taxi.main(frame).latestDistance == 0:
```

Comment:
First check that the return from taxi.main has a property called latestDistance, use OR.